### ISSUE TITLE
Migrations: MaxLength does not appear to work properly for sub types

### ISSUE THREAD

**Issue by dridgely** [dotnet/efcore#4238]
MaxLength does not appear to work properly for migrations of sub types. 

I have a sub type like:

```
public class SubType: SuperType {
    public string RelatedTypeKey {get;set;}
    public RelatedType Related {get;set;}
}
```

The model builder looks like:

```
modelBuilder.Entity<SubType>(entityBuilder => {
    entityBuilder.Property(subType => subType.RelatedTypeKey).MaxLength(50) 
});
```

The resulting model snapshot has:

```
b.Property<String>("RelatedTypeKey").HasAnnotation("MaxLength", 50)
```

The migration sql that results produces:

```
RelatedTypeKey nvarchar(max)
```

I've tried using attributes, as well, with the same result. This, of course, results in a failing SQL script because the foreign key length does not match the primary key length.

**Comment by rowanmiller** [comment]
Duplicate of https://github.com/aspnet/EntityFramework/issues/4146

**Comment by dridgely** [comment]
@rowanmiller I don't think this is a duplicate. You requested that I create this as a new issue at the end of #882. This does not work against 7.0.0-rc1-final. I've had to modify my model to move the foreign key to the super type in order to generate SQL that honors the MaxLength property.

**Comment by rowanmiller** [comment]
It looks like the same issue to me, you configure max length on a property defined on a derived type and it gets ignored (though maybe I am missing something?). When I wrote that comment on #882 I hadn't got to triaging #4146 - so I didn't realize someone else had already reported it.

BTW as a workaround you could edit the generated migration to specify the type to use for the column (rather than moving it to the super type).

**Comment by dridgely** [comment]
Sorry, thought you had closed it as a duplicate of #882. I need to pay closer attention. :) I've left it in my model so that I remember to change it when this is fixed - but thanks for the suggestion.
